system/jlib/jtrace.hpp

```diff
@@ -18,21 +18,197 @@
 #ifndef JTRACE_HPP
 #define JTRACE_HPP
 
+#undef UNIMPLEMENTED //opentelemetry defines UNIMPLEMENTED
+#include "opentelemetry/exporters/ostream/span_exporter_factory.h"
```

Comment:
Avoid these includes in the header file if at all possible...  Otherwise everything in the system is suddenly going to be including many more header files.